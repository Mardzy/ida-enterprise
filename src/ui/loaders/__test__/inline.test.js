import InlineDots from "../inline";

describe("The inline loader", () => {

  it("renders 3 dots by default", () => {
    const wrapper = mount(
      <InlineDots/>
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("renders 5 dots", () => {
    const wrapper = mount(
      <InlineDots amount={5}/>
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });

});