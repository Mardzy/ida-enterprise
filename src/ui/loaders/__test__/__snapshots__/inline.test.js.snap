// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The inline loader renders 3 dots by default 1`] = `
<InlineDots
  amount={3}
>
  <styled.span
    key="0"
  >
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": true,
            "lastClassName": "gQgMD",
            "rules": Array [
              "
  animation: 1s ",
              Keyframes {
                "id": "sc-keyframes-cFFqjv",
                "inject": [Function],
                "name": "cFFqjv",
                "rules": Array [
                  "@-webkit-keyframes cFFqjv{50%{color:transparent;}}",
                  "@keyframes cFFqjv{50%{color:transparent;}}",
                ],
                "toString": [Function],
              },
              " infinite;
  &:nth-child(1) {
    animation-delay: 0ms;
  }
  &:nth-child(2) {
    animation-delay: 250ms;
  }
  &:nth-child(3) {
    animation-delay: 500ms;
  }
",
            ],
          },
          "displayName": "styled.span",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "span",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <span
        className="sc-bdVaJa gQgMD"
      >
        .
      </span>
    </StyledComponent>
  </styled.span>
  <styled.span
    key="1"
  >
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": true,
            "lastClassName": "gQgMD",
            "rules": Array [
              "
  animation: 1s ",
              Keyframes {
                "id": "sc-keyframes-cFFqjv",
                "inject": [Function],
                "name": "cFFqjv",
                "rules": Array [
                  "@-webkit-keyframes cFFqjv{50%{color:transparent;}}",
                  "@keyframes cFFqjv{50%{color:transparent;}}",
                ],
                "toString": [Function],
              },
              " infinite;
  &:nth-child(1) {
    animation-delay: 0ms;
  }
  &:nth-child(2) {
    animation-delay: 250ms;
  }
  &:nth-child(3) {
    animation-delay: 500ms;
  }
",
            ],
          },
          "displayName": "styled.span",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "span",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <span
        className="sc-bdVaJa gQgMD"
      >
        .
      </span>
    </StyledComponent>
  </styled.span>
  <styled.span
    key="2"
  >
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": true,
            "lastClassName": "gQgMD",
            "rules": Array [
              "
  animation: 1s ",
              Keyframes {
                "id": "sc-keyframes-cFFqjv",
                "inject": [Function],
                "name": "cFFqjv",
                "rules": Array [
                  "@-webkit-keyframes cFFqjv{50%{color:transparent;}}",
                  "@keyframes cFFqjv{50%{color:transparent;}}",
                ],
                "toString": [Function],
              },
              " infinite;
  &:nth-child(1) {
    animation-delay: 0ms;
  }
  &:nth-child(2) {
    animation-delay: 250ms;
  }
  &:nth-child(3) {
    animation-delay: 500ms;
  }
",
            ],
          },
          "displayName": "styled.span",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "span",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <span
        className="sc-bdVaJa gQgMD"
      >
        .
      </span>
    </StyledComponent>
  </styled.span>
</InlineDots>
`;

exports[`The inline loader renders 5 dots 1`] = `
<InlineDots
  amount={5}
>
  <styled.span
    key="0"
  >
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": true,
            "lastClassName": "gQgMD",
            "rules": Array [
              "
  animation: 1s ",
              Keyframes {
                "id": "sc-keyframes-cFFqjv",
                "inject": [Function],
                "name": "cFFqjv",
                "rules": Array [
                  "@-webkit-keyframes cFFqjv{50%{color:transparent;}}",
                  "@keyframes cFFqjv{50%{color:transparent;}}",
                ],
                "toString": [Function],
              },
              " infinite;
  &:nth-child(1) {
    animation-delay: 0ms;
  }
  &:nth-child(2) {
    animation-delay: 250ms;
  }
  &:nth-child(3) {
    animation-delay: 500ms;
  }
",
            ],
          },
          "displayName": "styled.span",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "span",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <span
        className="sc-bdVaJa gQgMD"
      >
        .
      </span>
    </StyledComponent>
  </styled.span>
  <styled.span
    key="1"
  >
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": true,
            "lastClassName": "gQgMD",
            "rules": Array [
              "
  animation: 1s ",
              Keyframes {
                "id": "sc-keyframes-cFFqjv",
                "inject": [Function],
                "name": "cFFqjv",
                "rules": Array [
                  "@-webkit-keyframes cFFqjv{50%{color:transparent;}}",
                  "@keyframes cFFqjv{50%{color:transparent;}}",
                ],
                "toString": [Function],
              },
              " infinite;
  &:nth-child(1) {
    animation-delay: 0ms;
  }
  &:nth-child(2) {
    animation-delay: 250ms;
  }
  &:nth-child(3) {
    animation-delay: 500ms;
  }
",
            ],
          },
          "displayName": "styled.span",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "span",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <span
        className="sc-bdVaJa gQgMD"
      >
        .
      </span>
    </StyledComponent>
  </styled.span>
  <styled.span
    key="2"
  >
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": true,
            "lastClassName": "gQgMD",
            "rules": Array [
              "
  animation: 1s ",
              Keyframes {
                "id": "sc-keyframes-cFFqjv",
                "inject": [Function],
                "name": "cFFqjv",
                "rules": Array [
                  "@-webkit-keyframes cFFqjv{50%{color:transparent;}}",
                  "@keyframes cFFqjv{50%{color:transparent;}}",
                ],
                "toString": [Function],
              },
              " infinite;
  &:nth-child(1) {
    animation-delay: 0ms;
  }
  &:nth-child(2) {
    animation-delay: 250ms;
  }
  &:nth-child(3) {
    animation-delay: 500ms;
  }
",
            ],
          },
          "displayName": "styled.span",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "span",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <span
        className="sc-bdVaJa gQgMD"
      >
        .
      </span>
    </StyledComponent>
  </styled.span>
  <styled.span
    key="3"
  >
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": true,
            "lastClassName": "gQgMD",
            "rules": Array [
              "
  animation: 1s ",
              Keyframes {
                "id": "sc-keyframes-cFFqjv",
                "inject": [Function],
                "name": "cFFqjv",
                "rules": Array [
                  "@-webkit-keyframes cFFqjv{50%{color:transparent;}}",
                  "@keyframes cFFqjv{50%{color:transparent;}}",
                ],
                "toString": [Function],
              },
              " infinite;
  &:nth-child(1) {
    animation-delay: 0ms;
  }
  &:nth-child(2) {
    animation-delay: 250ms;
  }
  &:nth-child(3) {
    animation-delay: 500ms;
  }
",
            ],
          },
          "displayName": "styled.span",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "span",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <span
        className="sc-bdVaJa gQgMD"
      >
        .
      </span>
    </StyledComponent>
  </styled.span>
  <styled.span
    key="4"
  >
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": true,
            "lastClassName": "gQgMD",
            "rules": Array [
              "
  animation: 1s ",
              Keyframes {
                "id": "sc-keyframes-cFFqjv",
                "inject": [Function],
                "name": "cFFqjv",
                "rules": Array [
                  "@-webkit-keyframes cFFqjv{50%{color:transparent;}}",
                  "@keyframes cFFqjv{50%{color:transparent;}}",
                ],
                "toString": [Function],
              },
              " infinite;
  &:nth-child(1) {
    animation-delay: 0ms;
  }
  &:nth-child(2) {
    animation-delay: 250ms;
  }
  &:nth-child(3) {
    animation-delay: 500ms;
  }
",
            ],
          },
          "displayName": "styled.span",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "span",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <span
        className="sc-bdVaJa gQgMD"
      >
        .
      </span>
    </StyledComponent>
  </styled.span>
</InlineDots>
`;
